src/V3Builtin.cpp

```diff
@@ -0,0 +1,90 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: built-in packages and classes
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2004-2020 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+
+#include "V3Ast.h"
+#include "V3Builtin.h"
+#include "V3FileLine.h"
+#include "V3Parse.h"
+#include "V3ParseSym.h"
+
+namespace V3Builtin {
+void parseStdPackage(V3Parse& parser) {
+    string filename = v3Global.opt.hierTopDataDir() + "/builtin.v";
+    std::ofstream* ofp = nullptr;
+    ofp = V3File::new_ofstream(filename);
+    if (!ofp->fail()) {
+        *ofp << R"(
+        package std;
+            class process;
+                integer x;
+                function new();
+                    x = 42;
+                endfunction;
+                typedef enum { FINISHED, RUNNING, WAITING, SUSPENDED, KILLED } state;
+                extern static function process self();
+                extern function state status();
+                extern function void kill();
+                extern task await();
+                extern function void suspend();
+                extern function void resume();
+            endclass
+        endpackage
+        )" << std::endl;
+    }
+    if (ofp) {
+        ofp->close();
+        delete ofp;
+    }
+    parser.parseFile(new FileLine(FileLine::commandLineFilename()), filename, false,
+                     "Cannot find file with built-in definitions: ");
+}
+
+void defineExterns(AstNetlist* rootp, V3ParseSym& parseSyms) {
+    auto* pkgp = rootp->modulesp();
+    while (pkgp->name() != "std") { pkgp = VN_CAST(pkgp->nextp(), NodeModule); }
+    parseSyms.pushNew(pkgp);
+    auto* classp = parseSyms.symCurrentp()->findIdFallback("process")->nodep();
+    auto* memberp = VN_CAST(classp, Class)->membersp();
+    while (memberp) {
+        AstNodeFTask* ftaskp = nullptr;
+        if (VN_IS(memberp, Task)) {
+            ftaskp = new AstTask(rootp->fileline(), memberp->name(), nullptr);
+        } else if (memberp->name() == "status") {
```

Comment:
It's generally not obvious to me what this code is doing that you couldn't do by changing the parsed code (make it not be extern but real definitions) - but maybe there will be more here eventually.

Reply:
The parsed code is all `extern`s because we can't really implement any of these methods in Verilog, we'll probably want to call some code from `verilated.h`. For now it's adding empty definitions because otherwise Verilator would complain about missing function definitions.

Reply:
Makes sense, my point was only for now you could just stub them in the verilog and save some work.  If you're not going to submit this partially as-is but keep going it's fine to keep as-is.

Reply:
I don't think we need to merge it for now (@tgorochowik ?), as it doesn't really provide any useful functionality yet.
I figured it would be good to have the task/func creation here as a starting point for whoever will be implementing these functions (it may or may not be me).

Reply:
I personally think it would be useful to keep it, but I am okay with keeping that on a separate branch for now (maybe we could create a separate issue/pr so it's easier to pick it up later?)